dipy/core/gradients.py

```diff
@@ -674,7 +674,87 @@ def round_bvals(bvals, bmag=None):
     return b.round() * (10 ** bmag)
 
 
-def unique_bvals(bvals, bmag=None, rbvals=False):
+def unique_bvals_tolerance(bvals, tol=20):
+    """ Gives the unique b-values of the data, within a tolerance gap
+
+    The b-values must be regrouped in clusters easily separated by a
+    distance greater than the tolerance gap. If all the b-values of a
+    cluster fit within the tolerance gap, the highest b-value is kept.
+
+    Parameters
+    ----------
+    bvals : ndarray
+        Array containing the b-values
+
+    tol : int
+        The tolerated gap between the b-values to extract
+        and the actual b-values.
+
+    Returns
+    ------
+    ubvals : ndarray
+        Array containing the unique b-values using the median value
+        for each cluster
+    """
+    b = np.unique(bvals)
+    ubvals = []
+    i = 0
+    lower_part = np.where(b <= b[i] + tol)[0]
```

Comment:
For clarity, Why you do not put directly `b[0]`? What is the goal to initialize `i` here?